src/ExpressionEvaluator/CSharp/Test/ResultProvider/DynamicTests.cs

```diff
@@ -0,0 +1,263 @@
+                EvalResult("Q", "null", "object[]", "d.Q"),
+                EvalResult("R (Base<object, object, object, object>)", "null", "object", "((Base<object, object, object, object>)d).R"),
+                EvalResult("R", "null", "dynamic[] {object[]}", "d.R"),
+                EvalResult("S (Base<object, object, object, object>)", "null", "object", "((Base<object, object, object, object>)d).S"),
+                EvalResult("S", "null", "object[]", "d.S"));
+        }
+
+    [Fact]
+        public void ArrayElement()
+        {
+            var value = CreateDkmClrValue(new object[1]);
+            var rootExpr = "d";
+            var evalResult = FormatResult(rootExpr, rootExpr, value, declaredType: new DkmClrType((TypeImpl)typeof(object[])), declaredTypeInfo: new[] { false, true });
+            Verify(evalResult,
+                EvalResult(rootExpr, "{object[1]}", "dynamic[] {object[]}", rootExpr, DkmEvaluationResultFlags.Expandable));
+            var children = GetChildren(evalResult);
+            Verify(children,
+                EvalResult("[0]", "null", "dynamic {object}", "d[0]"));
+        }
+
+        [Fact]
+        public void TypeVariables()
+        {
+            var intrinsicSource =
+@".class private abstract sealed beforefieldinit specialname '<>c__TypeVariables'<T,U,V>
+{
+  .method public hidebysig specialname rtspecialname instance void .ctor() { ret }
+}";
+            ImmutableArray<byte> assemblyBytes;
+            ImmutableArray<byte> pdbBytes;
+            CommonTestBase.EmitILToArray(intrinsicSource, appendDefaultHeader: true, includePdb: false, assemblyBytes: out assemblyBytes, pdbBytes: out pdbBytes);
+            var assembly = ReflectionUtilities.Load(assemblyBytes);
+            var reflectionType = assembly.GetType(ExpressionCompilerConstants.TypeVariablesClassName).MakeGenericType(new[] { typeof(object), typeof(object), typeof(object[]) });
+            var value = CreateDkmClrValue(value: null, type: reflectionType, valueFlags: DkmClrValueFlags.Synthetic);
+            var evalResult = FormatResult("typevars", "typevars", value, new DkmClrType((TypeImpl)reflectionType), new[] { false, true, false, false, true });
+            Verify(evalResult,
+                EvalResult("Type variables", "", "", null, DkmEvaluationResultFlags.Expandable | DkmEvaluationResultFlags.ReadOnly, DkmEvaluationResultCategory.Data));
+            var children = GetChildren(evalResult);
+            Verify(children,
+                EvalResult("T", "dynamic", "dynamic", null, DkmEvaluationResultFlags.ReadOnly, DkmEvaluationResultCategory.Data),
+                EvalResult("U", "object", "object", null, DkmEvaluationResultFlags.ReadOnly, DkmEvaluationResultCategory.Data),
+                EvalResult("V", "dynamic[]", "dynamic[]", null, DkmEvaluationResultFlags.ReadOnly, DkmEvaluationResultCategory.Data));
+        }
+    }
```

Comment:
Huh, I could have sworn I'd already added such tests.  I must have since I commented out the asserts that only hold for valid inputs.